torch/tensor.py

```diff
@@ -745,10 +910,33 @@ def grad(self):
 
     @grad.setter
     def grad(self, new_grad):
+        relevant_args = (self,)
+        from torch._overrides import has_torch_function, handle_torch_function
+        if type(self) is not Tensor and has_torch_function(relevant_args):
+            return handle_torch_function(Tensor.grad.__set__, relevant_args, self, new_grad)
         self._grad = new_grad
 
     @grad.deleter
     def grad(self):
+        relevant_args = (self,)
+        from torch._overrides import has_torch_function, handle_torch_function
+        if type(self) is not Tensor and has_torch_function(relevant_args):
+            return handle_torch_function(Tensor.grad.__delete__, relevant_args, self)
         del self._grad
+    
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        if not all(issubclass(cls, t) for t in types):
+            return NotImplemented
+
+        with _C.DisableTorchFunction():
```

Comment:
Do you expect users are going to use `_C.DisableTorchFunction`? If so, we need a more user-friendly API name for it.

Reply:
Probably not, see https://github.com/pytorch/pytorch/pull/37091#discussion_r433799872 which covers the majority of use-cases.